src/assignment/helpers/insert-at-position.ts

```diff
@@ -0,0 +1,17 @@
+type Positioned<T> = T & { id: string; position: number };
+
+const sortByPositionAsc = (a: Positioned<unknown>, b: Positioned<unknown>) =>
+	a.position - b.position;
+
+export function spliceByPosition<T>(list: Positioned<T>[], item: Positioned<T>): Positioned<T>[] {
+	const sorted = [...list.sort(sortByPositionAsc)];
+
+	sorted.splice(item.position, 0, item);
+
+	return sorted
+		.map((item, i) => ({
+			...item,
+			position: i,
+		}))
+		.sort(sortByPositionAsc);
```

Comment:
no need to sort again, since you just ran (map) through every item and set the position to be the index. Index will go from 0 -> n